src/t8_vtk/t8_vtk_writer.hxx

```diff
@@ -230,105 +216,57 @@ class vtk_writer {
     const int num_node = t8_get_number_of_vtk_nodes (element_shape, curved_flag);
 
-    if (curved_flag == 0) {
-      switch (element_shape) {
-      case T8_ECLASS_VERTEX:
-        pvtkCell = vtkSmartPointer<vtkVertex>::New ();
-        break;
-      case T8_ECLASS_LINE:
-        pvtkCell = vtkSmartPointer<vtkLine>::New ();
-        break;
-      case T8_ECLASS_QUAD:
-        pvtkCell = vtkSmartPointer<vtkQuad>::New ();
-        break;
-      case T8_ECLASS_TRIANGLE:
-        pvtkCell = vtkSmartPointer<vtkTriangle>::New ();
-        break;
-      case T8_ECLASS_HEX:
-        pvtkCell = vtkSmartPointer<vtkHexahedron>::New ();
-        break;
-      case T8_ECLASS_TET:
-        pvtkCell = vtkSmartPointer<vtkTetra>::New ();
-        break;
-      case T8_ECLASS_PRISM:
-        pvtkCell = vtkSmartPointer<vtkWedge>::New ();
-        break;
-      case T8_ECLASS_PYRAMID:
-        pvtkCell = vtkSmartPointer<vtkPyramid>::New ();
-        break;
-      default:
-        SC_ABORT_NOT_REACHED ();
-      }
-    }
-    else { /* curved_flag != 0 */
-      switch (element_shape) {
-      case T8_ECLASS_VERTEX:
-        pvtkCell = vtkSmartPointer<vtkVertex>::New ();
-        break;
-      case T8_ECLASS_LINE:
-        pvtkCell = vtkSmartPointer<vtkQuadraticEdge>::New ();
-        break;
-      case T8_ECLASS_QUAD:
-        pvtkCell = vtkSmartPointer<vtkQuadraticQuad>::New ();
-        break;
-      case T8_ECLASS_TRIANGLE:
-        pvtkCell = vtkSmartPointer<vtkQuadraticTriangle>::New ();
-        break;
-      case T8_ECLASS_HEX:
-        pvtkCell = vtkSmartPointer<vtkQuadraticHexahedron>::New ();
-        break;
-      case T8_ECLASS_TET:
-        pvtkCell = vtkSmartPointer<vtkQuadraticTetra>::New ();
-        break;
-      case T8_ECLASS_PRISM:
-        pvtkCell = vtkSmartPointer<vtkQuadraticWedge>::New ();
-        break;
-      case T8_ECLASS_PYRAMID:
-        pvtkCell = vtkSmartPointer<vtkQuadraticPyramid>::New ();
-        break;
-      default:
-        SC_ABORT_NOT_REACHED ();
-      }
-    }
+    /* Create an array with the correct number of indices per cell */
+    vtkIdType *vecCellIds = new vtkIdType[num_node];
 
     /* Compute the coordinates of the element/tree. */
     double *coordinates = T8_ALLOC (double, 3 * num_node);
 
     grid_element_to_coords (grid, itree, element, curved_flag, coordinates, num_node, element_shape);
 
+    vtkIdType ptId = -1;
     for (int ivertex = 0; ivertex < num_node; ivertex++, (*point_id)++) {
       const size_t offset_3d = 3 * ivertex;
+
       /* Insert the point in the points array. */
-      points->InsertNextPoint (coordinates[offset_3d], coordinates[offset_3d + 1], coordinates[offset_3d + 2]);
-      pvtkCell->GetPointIds ()->SetId (ivertex, *point_id);
+      double vtkCoords[3] = { coordinates[offset_3d], coordinates[offset_3d + 1], coordinates[offset_3d + 2] };
+      if (mergePoints) {
+        points->InsertUniquePoint (vtkCoords, ptId);
+      }
```

Comment:
I added the possibility to allow duplicated point insertion. The default behaviour is to merge duplicated points